src/Repository/RepositoryStack.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Repository;
+
+/**
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+class RepositoryStack
```

Comment:
Yeah this is what i mean, sorry if it wasn't clear.
This PR only proposes a solution for managing multiple repositories not for managing them differently.
But as i said in your PR, the EventDispatcher is imo what we need and we could create a features or listeners or whatever property in the Repository class to enable or disable a listeners based on the current repository.

Of course this PR is not as complete as yours (the event management could be used here too) and i won't have the time before wednesday to do big changes so i let @WouterJ and @weaverryan decide which approach they prefer.

Thanks for your review and your PR ☺